File: knowing.test/src/main/scala/knowing/test/Activator.scala
package knowing.test

import de.lmu.ifi.dbs.knowing.core.service._
import de.lmu.ifi.dbs.knowing.core.factory.TFactory
import de.lmu.ifi.dbs.knowing.core.util.OSGIUtil

import knowing.test.loader._
import knowing.test.processor._
import knowing.test.filter._

import org.osgi.framework.ServiceReference
import org.osgi.framework.ServiceRegistration
import org.osgi.framework.BundleContext
import org.osgi.framework.BundleActivator
import org.osgi.util.tracker.ServiceTrackerCustomizer
import org.osgi.util.tracker.ServiceTracker

class Activator extends BundleActivator {

  def start(context: BundleContext) = {
  }

  def stop(context: BundleContext) = {
  }

}

object Activator {

  private var context: BundleContext = null

  def getContext(): BundleContext = context
}

File: de.lmu.ifi.dbs.knowing.debug.core/src/de/lmu/ifi/dbs/knowing/debug/core/internal/Activator.java
package de.lmu.ifi.dbs.knowing.debug.core.internal;

import java.nio.file.Path;
import java.nio.file.Paths;

import org.osgi.framework.BundleActivator;
import org.osgi.framework.BundleContext;
import org.osgi.framework.ServiceRegistration;

import akka.actor.TypedActor;
import akka.actor.TypedActorFactory;

import de.lmu.ifi.dbs.knowing.core.factory.UIFactory;

public class Activator implements BundleActivator {

	public static final String PLUGIN_ID = "de.lmu.ifi.dbs.knowing.debug.core";
	
	private static BundleContext context;

	static BundleContext getContext() {
		return context;
	}

	/*
	 * (non-Javadoc)
	 * @see org.osgi.framework.BundleActivator#start(org.osgi.framework.BundleContext)
	 */
	public void start(BundleContext bundleContext) throws Exception {
		Activator.context = bundleContext;
	}

	/*
	 * (non-Javadoc)
	 * @see org.osgi.framework.BundleActivator#stop(org.osgi.framework.BundleContext)
	 */
	public void stop(BundleContext bundleContext) throws Exception {
		Activator.context = null;
	}

}
